dm/syncer/syncer.go

```diff
@@ -1948,16 +1952,10 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 				currentLocation.Suffix = 0
 				s.isReplacingOrInjectingErr = false
 				s.locations.reset(currentLocation)
-				if s.errOperatorHolder.IsInject(startLocation) {
-					s.errOperatorHolder.SetHasAllInjected(startLocation)
-					// reset event as startLocation, avoid to be marked in checkpoint
-					currentLocation.Position.Pos = startLocation.Position.Pos
-					err = s.streamerController.RedirectStreamer(tctx, startLocation)
-				} else {
-					err = s.streamerController.RedirectStreamer(tctx, currentLocation)
-				}
-				if err != nil {
-					return err
+				if !s.errOperatorHolder.IsInject(startLocation) {
+					if err = s.streamerController.RedirectStreamer(tctx, currentLocation); err != nil {
```

Comment:
seems we can remove the outed commnet `also redirect stream to next event` in L1950, now only redirect stream when replace

Reply:
ok